src/qt/transactiontablemodel.cpp

```diff
@@ -167,10 +167,28 @@ class TransactionTablePriv
         case CT_UPDATED:
             // Miscellaneous updates -- nothing to do, status update will take care of this, and is only computed for
             // visible transactions.
+            for (int i = lowerIndex; i < upperIndex; i++) {
+                TransactionRecord *rec = &cachedWallet[i];
+                rec->status.needsUpdate = true;
+            }
+            Q_EMIT parent->dataChanged(parent->index(lowerIndex, TransactionTableModel::Status), parent->index(upperIndex, TransactionTableModel::Status));
             break;
         }
     }
 
+    void updateAddressBook(const QString& address, const QString& label, bool isMine, const QString& purpose, int status)
+    {
+        std::string address2 = address.toStdString();
+        int index = 0;
+        for (auto& rec : cachedWallet) {
+            if (rec.strAddress == address2) {
+                rec.status.needsUpdate = true;
+                Q_EMIT parent->dataChanged(parent->index(index, TransactionTableModel::Status), parent->index(index, TransactionTableModel::Status));
```

Comment:
Can't we avoid triggering yet another update and just do `rec.status.label = label;` here (and also drop corresponding code from `TransactionRecord::updateStatus()`)?

Reply:
This is what I initially did, but then realized that we would still have to initialize the label to an initial value. And we also have to re-trigger filtering in `TransactionFilterProxy` as this TX now might become visible or invisible if label filtering is done.